### DISCUSSION TITLE
SEI layer models

### DISCUSSION THREAD

**Discussion by optibikash** [pybamm-team/PyBaMM#2148] [General]
Can someone explain the difference between the different classes of SEI defined in https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/interface/sei/sei_growth.py.
I am more interested to understand class pybamm.sei.ReactionLimited and class pybamm.sei.EcReactionLimited. However, it is good to have a detailed differentiation between all sub-classes of SEI. I want to understand with respect the electrochemical reactions or the processes defined as per the classes. The mathematical equations are defined, however, the definition given in the docs doesn't completely explain. For example, for the class reactionlimited, it is mentioned: "Class for reaction limited SEI growth". The definition doesn't explain the process, assumptions or anything else that helps to understand more.

**Comment by brosaplanella** [comment]
The `EC reaction limited` model follows from the [article by Yang et al 2017](https://www.sciencedirect.com/science/article/pii/S0378775317307619) while the other models are taken from [Scott Marquis' thesis](https://ora.ox.ac.uk/objects/uuid:8afdcc34-cc42-48ba-b316-96a6d0f33a45). These references provide a more detailed description of the models.

**Reply by optibikash** [reply]
Thanks, Ferran. It's worth refering to the paper. Glad to hear from you.

**Comment by robin2715** [comment]
Hello all

First at all, thanks for sharing this code. It is highly appreciated!
I have a follow up question with respect to this. I am comparing the implemented "EC reaction limited"  with what is described by Yang et al 2017, but I cannot come to the same equations of j_sei.

j_sei is originally implemented by (sey_growth.py, line 110):
j_sei = - C_sei_ec * c_ec * exp()

Then, the following additional equations are used:
1. C_sei_ec definition (from lithium_ion_parameters.py, line 379)      
   C_sei_ec =    F * k_sei_dim * c_ec_0_dim / j_scale  * ( exp( -(F * (U_ref - U_sei_dim)/ (2 * R * T_ref) ) );
			
2. exp definition
   exp = exp(-0.5 * (delta_phi - j * L_sei * R_sei));

3. R_sei definition (lithium_ion_parameters.py line 348)
    R_sei = F*j_scale*R_sei_dimensional*L_sei_0_dim/ R/ T_ref ;

4. j_scale definition (lithium_ion_parameters.py line 722 (porous electrode))
    j_scale = i_typ / (a_typ * L_x)

Replacing all back into j_sei equation gives:
j_sei = -  F * k_sei_dim * c_ec_0_dim * c_ec */ j_scale *  
    exp{ -F / (2 * R * T_ref) * (U_ref - U_sei_dim) -0.5 * (delta_phi - F/(R* T_ref)*j * j_scale * L_sei * L_sei_0_dim *R_sei_dimensional) } 

In the paper, j_sei is defined by:
j_sei = - a* F * k_sei_dim *  c_ec *  
    exp{ F / (2 * R * T_ref) * (U_sei) - 0.5 * F/(R* T_ref) * (delta_phi - j*R_sei_dimensional/a ) } 


Questions
1. Are the terms "L_sei_0_dim" and "c_ec_0_dim" the scaling values for SEI thickness and concentration, respectively?
2. Does "delta_phi" already include the term F/(2*R* T_ref)? I could not locate the equation where is actually computed. 
3.  U_sei = -(U_ref - U_sei_dim) (?)
4. I am not sure how are j and j_scale used. I noticed that a_typ is missing  in the implemented equation. If the definition of j_scale is replaced in j_sei, then a_typ will be introduced but it will also introduce L_x to the equation. It seems to me that a_typ is simply missing from the equations. 
5. L_sei does not appear in the exponential part in the paper. 

Where
c_ec =  EC surface concentration [dimensionless]
L_sei = Total SEI thickness [?]
L_sei_0_dim = Initial outer + inner SEI thickness [m]
R_sei = SEI resistivity [-]
R_sei_dimensional = SEI resistivity [Ohm.m]
j_sei = local current density of sei formation [?]. 
c_ec_0_dim = EC initial concentration in electrolyte [mol.m-3]
k_sei_dim = SEI kinetic rate constant [m.s-1]
T_ref = "Reference temperature [K]"
j_scale = Scale for interfacial current density in [A/m2]
U_ref = Dimensional open-circuit potential [V]
U_sei_dim = "SEI open-circuit potential [V]"
delta_phi = phi_s - phi_e, Lithium metal interface surface potential difference [?]
F =  Farady constant [C/mol] 
R = Gas constant [J/K/mol]
j = electrode total interfacial current density [?]
J_sei = SEI interfacial current density [?]
a_typ = specific surface area [m^-1]
L_x = Total distance between current collectors [m]

**Comment by valentinsulzer** [comment]
> Are the terms "L_sei_0_dim" and "c_ec_0_dim" the scaling values for SEI thickness and concentration, respectively?

Yes

> Does "delta_phi" already include the term F/(2R T_ref)? I could not locate the equation where is actually computed.

delta_phi is scaled by F/(RT_ref)


> U_sei = -(U_ref - U_sei_dim) (?)

Yes

> I am not sure how are j and j_scale used. I noticed that a_typ is missing in the implemented equation. If the definition of j_scale is replaced in j_sei, then a_typ will be introduced but it will also introduce L_x to the equation. It seems to me that a_typ is simply missing from the equations.

In the paper di/dx = j and j = a*(...) (j has units A/m^3), and in pybamm di/dx = a*j and j = (...) (j has units A/m^2). It's just a different convention.

> L_sei does not appear in the exponential part in the paper.

In the paper delta_film (=L_sei) is included in the definition of R_film in eq (15)

**Comment by robin2715** [comment]
Thanks for your answers. Questions 1-4 are clear. However, question 5 still does not completely fit :

Exponential terms in pybamm
ExpTermPybamm = -F/(R* T_ref)*(j * j_scale) * (L_sei * L_sei_0_dim) * R_sei_dimensional

Exponential term in the paper (using Pybamm's notation)
ExpTermPaper = -F/(R* T_ref)*(j * j_scale) * (R_sei * R_sei_dimensional)

Eq. 15 in the paper (using Pybamms notation)
R_sei * R_sei_dimensional = epsilon_s * (L_sei * L_sei_0_dim) / kappa_inner_dimensional

Replacing Eq. 15 in ExpTermPaper 
ExpTermPaper = -F/(R* T_ref)*(j * j_scale) * epsilon_s * (L_sei * L_sei_0_dim) / kappa_inner_dimensional


Questions
1. As you can see, ExpTermPaper   and ExpTermPybamm  still don't match. epsilon_s and kappa_inner_dimensional are not present in Pybamm's implementation. I also cannot locate an epsilon related to SEI only. Am I missing something here?
2. R_film_paper in the paper has units of [Ohm*m2].  Is it correct that in pybamm the units of R_sei_dimensional are [Ohm*m] (lithium_ion_parameters.py line 130)?.
2. delta_film  is a function of SEI growth and Lithium plating (Eq. 14 in the paper). Is it correct that Lithium plating is not considered here?

Assumptions: 
1. j_paper = j * j_scale / a_typ
2. R_film_paper = R_sei * R_sei_dimensional
3. delta_film_paper = L_sei * L_sei_0_dim

New variables
kappa_inner_dimensional = Inner SEI electron conductivity [S.m-1]
epsilon_s = electrode active material volume fraction (should belong to SEI?))

Thanks for your time!

**Reply by robin2715** [reply]
@tinosulzer any thoughts on this?

**Reply by redchang111** [reply]
Hello, I have the same question with you. Do you solve it?

**Reply by robin2715** [reply]
No official solution here. What I am trying to do now is to make a 'clean' implementation of the models with what I think is correct.
